packages/shorebird_cli/lib/src/artifact_builder/artifact_builder.dart

```diff
@@ -93,22 +96,36 @@ class ArtifactBuilder {
         'build',
         'appbundle',
         '--release',
+        '-v',
         if (flavor != null) '--flavor=$flavor',
         if (target != null) '--target=$target',
         if (targetPlatformArgs != null) '--target-platform=$targetPlatformArgs',
         ...args,
       ];
 
-      final result = await process.run(
+      final spawnedProcess = await process.start(
         executable,
         arguments,
         runInShell: true,
         environment: base64PublicKey?.toPublicKeyEnv(),
       );
 
-      if (result.exitCode != ExitCode.success.code) {
+      final sub = spawnedProcess.stdout.map(utf8.decode).listen((event) {
+        progressUpdater?.onLog(event);
+      });
+
+      final stderr = StringBuffer();
+      final stderrSub =
+          spawnedProcess.stderr.map(utf8.decode).listen(stderr.write);
+
+      final exitCode = await spawnedProcess.exitCode;
+
+      await sub.cancel();
+      await stderrSub.cancel();
+
+      if (exitCode != ExitCode.success.code) {
```

Comment:
I'm wondering if this is the right level to make this change. To extend this to other build types, we'll need to add this exact same ~20 lines of code to every `build[ArtifactType]` method in this file. I wonder if a slightly modified version of this in https://github.com/shorebirdtech/shorebird/blob/main/packages/shorebird_cli/lib/src/shorebird_process.dart#L109 would be easier to write/test/maintain?

Reply:
Hum, that is an interesting idea! I will try to come up with something like that!

Reply:
I refactored a lot the PR and it should now look like more to what you suggested here, lmk wdyt